source-container-build/app/source_build.py

```diff
@@ -391,9 +391,23 @@ def make_source_archive(
     repo_info = get_repo_info(source_dir)
     name_sha = f"{repo_info['name']}-{repo_info['last_commit_sha']}"
     output_archive = f"{source_archive_dir}/{name_sha}.tar.gz"
-    git_cmd = ["git", "archive", "--prefix", name_sha + "/", "--output", output_archive, "HEAD"]
-    log.debug("Generate source archive %r", git_cmd)
-    run(git_cmd, check=True, cwd=source_dir)
+
+    restore_cmd = ["git", "restore", "."]
```

Comment:
With this change, does the task now undo modifications in the source repo? AFAIK previously, the task would generate a clean archive but leave any modifications in the worktree untouched. IMO it'd be better to avoid this change in behavior unless its explicitly desired

Reply:
Yeah, you're right. It's not explicitly desired to modify the worktree - but the intent is to generate the archive from the original committed content.

Reply:
> the intent is to generate the archive from the original committed content.

+1, we should do that. Maybe we can `git stash push` -> `git stash pop` the changes instead of discarding them?

Reply:
Since source-build task runs in parallel with other tasks, `stash push -> stash pop` might cause conflict with potentially, although it is not recommended to make changes or write new artifacts inside the `source/`. `git worktree` can help without changing the `source/`.